lib/note-revisions/index.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+
+import NotePreview from '../components/note-preview';
+import TagChip from '../components/tag-chip';
+import { noteCanonicalTags } from '../state/selectors';
+import { tagHashOf } from '../utils/tag-hash';
+
+import type * as S from '../state';
+import type * as T from '../types';
+
+type OwnProps = {
+  noteId: T.EntityId;
+  note?: T.Note;
+};
+
+type StateProps = {
+  tags: Array<{ name: T.TagName; deleted: boolean }>;
+  noteId: T.EntityId | null;
+  note: T.Note | null;
+};
+
+type Props = OwnProps &
+  StateProps &
+  Pick<React.HTMLProps<HTMLDivElement>, 'aria-hidden'>;
+
+export class NoteRevisions extends Component<Props> {
+  static displayName = 'NoteRevisions';
+
+  render() {
+    const { note, noteId, tags, 'aria-hidden': ariaHidden } = this.props;
+
+    return (
+      <div aria-hidden={ariaHidden} className="note-revisions">
+        <NotePreview noteId={noteId} note={note} />
+        <div className="tags">
+          {tags.map(({ name, deleted }) => (
+            <TagChip
+              key={name}
+              tagName={name}
+              interactive={false}
+              deleted={deleted}
+            />
+          ))}
+        </div>
+      </div>
+    );
+  }
+}
+
+const mapStateToProps: S.MapState<StateProps, OwnProps> = (state, props) => {
+  const noteId = props.noteId ?? state.ui.openedNote;
+  const note = props.note ?? state.data.notes.get(noteId);
+  const restoreDeletedTags = state.ui.restoreDeletedTags;
+
+  const tags = noteCanonicalTags(state, note)
+    .map((tagName) => {
+      const tagHash = tagHashOf(tagName);
+      return { name: tagName, deleted: !state.data.tags.has(tagHash) };
+    })
+    .filter((tag) => {
+      return restoreDeletedTags || !tag.deleted;
+    });
+
+  return {
+    tags,
+    noteId,
+    note,
+  };
+};
```

Comment:
based on our current use of the revisions screen I think this is fine, but I did want to point out that this triggers a re-render unconditionally because of how we're creating new copies of the lists in `mapStateToProps`.

if in the future we want to address this, or if we end up with glitchy rendering, we can memoize it here or as a selector, or since this is all relatively stateless computation we can move that into the component itself where it will only run when the props change vs. on every update to the app, even unrelated updates. well, I guess `noteCanonicalTags` is creating a new copy as well - eventually we might want to look at that.

this is for information purposes only and to leave a trace of the conversation; it's not a blocker for this PR